### ISSUE TITLE
Setting __Zone_disable_ZoneAwarePromise to true throws error

### ISSUE THREAD

**Issue by RatkoR** [angular/angular#37349]
# 🐞 bug report

### Affected Package

zone.js

### Is this a regression?

I tried with angular 8 on stackblitz and it also throws Error.

### Description

Setting `__Zone_disable_ZoneAwarePromise = true;` should disable zonejs patching of Promise.then. Instead, it throws `Promise has beed overwritten` error and stops.

## 🔬 Minimal Reproduction

Just add `__Zone_disable_ZoneAwarePromise = true;` to index.html as per [documentation](https://github.com/angular/angular/blob/master/packages/zone.js/MODULE.md).

https://stackblitz.com/edit/angular-bazsff

## 🔥 Exception or Error
<pre><code>
Error: Zone.js has detected that ZoneAwarePromise `(window|global).Promise` has been overwritten.
</code></pre>

from [packages/zone.js/lib/zone.ts::assertZonePatched](https://github.com/angular/angular/blob/master/packages/zone.js/lib/zone.ts#L726).

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 9.1.6
Node: 10.15.3
OS: linux x64

Angular: 9.1.7
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, platform-server
... router, upgrade
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.901.6
@angular-devkit/build-angular     0.901.6
@angular-devkit/build-optimizer   0.901.6
@angular-devkit/build-webpack     0.901.6
@angular-devkit/core              9.1.6
@angular-devkit/schematics        9.1.6
@angular/cdk                      9.2.1
@angular/cli                      9.1.6
@ngtools/webpack                  9.1.6
@nguniversal/builders             9.1.1
@nguniversal/common               9.1.1
@nguniversal/express-engine       9.1.1
@schematics/angular               9.1.6
@schematics/update                0.901.6
rxjs                              6.5.5
typescript                        3.8.3
webpack                           4.42.0

zone.js: 0.10.3
</code></pre>

**Anything else relevant?**

Polyfills.js states that only 3 variables have affect:
- __Zone_disable_requestAnimationFrame
- __Zone_disable_on_property
- __zone_symbol__UNPATCHED_EVENTS

If this is true, than maybe you only need to fix zonejs documentation, because it mentions a lot of other flags, __Zone_disable_ZoneAwarePromise being one of them.

**Comment by JiaLiPassion** [comment]
@RatkoR, this is by design, `angular` need `zone.js` to patch `Promise`, so if the promise is not `ZoneAwarePromise`, `Angular`  will throw this error.
`zone.js` does support disable `Promise` patch, but that is the feature of `zone.js`, you can use the `zone.js` not only in `angular`, but also in other javascript applications.
So could you give me more information about why you want `Promise` patch be disabled?

**Comment by RatkoR** [comment]
Aha ok.

I thought it can be disabled in angular also. We needed to load an external script and it broke our angular app. I experimented a bit and wanted to see how this external script behaves if zonejs doesn't patch promise.

Since only a subset of `__Zone_disable_*` flags can be used, maybe docs could be updated a bit?

The external script was messing with promise:

```
if (window.Promise && window.Proxy) {
    var e = {
        construct: function (t, e) {
            return Reflect.construct(t, e);
        },
    };
    window.Promise = new Proxy(window.Promise, e);
}
```

Open up angular.io site and paste this code into console in dev tools. It'll give you `ERROR RangeError: Maximum call stack size exceeded at w (zone-evergreen.js:728)`.

Firefox and chrome produce an error while chrome pumps up CPU and memory just dies eventually (without giving any errors).

Could zonejs handle this in any way?
